optaweb-vehicle-routing-frontend/cypress/integration/fromLocationsToRoute.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describe('Locations can be added and route is computed', () => {
+
+  /**
+   * Adds a location by searching for a city of a given name.
+   * @param {city name} name 
+   */
+  const addCity = (name) => {
+    cy.get('[data-cy=demo-location-text-input]').type(name);
+    cy.get('[data-cy=demo-location-item-button-1]').click();
+  };
+
+  /**
+   * Clears locations by clicking on a 'Clear' button.
+   */ 
```

Comment:
Could you please configure your VSCode to trim trailing whitespace and always insert a final newline?

Final newline instructions: https://stackoverflow.com/questions/44704968/visual-studio-code-insert-new-line-at-the-end-of-files.